src/PHPMailer.php

```diff
@@ -1237,7 +1237,7 @@ public static function parseAddresses($addrstr, $useimap = true)
                     $name = trim($name);
                     if (static::validateAddress($email)) {
                         //If this name is encoded, decode it
-                        if (preg_match('/^=\?.*\?=$/', $name)) {
+                        if (extension_loaded('mbstring') && preg_match('/^=\?.*\?=$/', $name)) {
```

Comment:
It's worth avoiding `extension_loaded` because it's sometimes disabled in shared hosting environments. It's safer to check for defined constants, for example using `defined('MB_CASE_UPPER')`. See the same thing is done for detecting OpenSSL.

Reply:
Agreed. (Though seriously what are they thinking ? Security by obscurity is not security.)
Either way, in the same function, same thing, but in the different "arm" of the logic, it is done in the same way using `extension_loaded`, so for consistency, I'd say that in that case *both* need to be changed.

https://github.com/PHPMailer/PHPMailer/blob/5cbd8f9f62ffe03f120b733c50c7bdc4b655b929/src/PHPMailer.php#L1205-L1212

Note: I did already consider changing to a `function_exists()` originally, but as the second bugfix to be pulled after this one introduces the use of another Mbstring function, `function_exists()` did not seem appropriate (unless used twice as individual functions can also be disabled by hosts via the `disable_functions` ini setting).

Reply:
Oh I know – there is little about shared hosting that's not crappy! Yes, other instances of it should be changed too. Only thing to look out for is in what PHP versions the constants you check for were introduced, as these checks have to work all the way back to 5.5.

Reply:
> Yes, other instances of it should be changed too.

Will do.

> ... these checks have to work all the way back to 5.5.

Fully aware of that, that is what the upcoming bugfix 2 is about (this function currently does not work on PHP 5.5).

Reply:
I've added an extra commit to this PR to fix this up.